### ISSUE TITLE
Add generic click function

### ISSUE THREAD

**Issue by petelacey** [germsvel/phoenix_test#140]
Maybe it's already here and I'm missing something, but can a generic `click` function that returns a session be added to PhoenixTest, similar to `click_link` and `click_button` but for any kind of element?

In may case I have divs with phx-click events on them (code smell?).

```elixir
test "admin can edit a user", %{conn: conn} do
  conn
  |> visit("/")
  |> click("div[phx-click=edit-user]", "Edit")
  |> fill_in("Name", with: "Aragorn")
  |> click_button("Create")
  |> assert_has(".user", text: "Aragorn")
end
```

**Comment by germsvel** [comment]
@petelacey thanks for opening this issue. 

I do hope to introduce a more generic `click` function that would work with other helpers like `button` or `link` or perhaps a base `element`. But we don't have that now. 

Now you ask if it's a code smell. 

I don't know how your app is designed. _Technically_, LiveView lets you put `phx-click` on any element, so putting them on a `div` is fine. 

That being said, looking at your example, it seems like you're trying to click on an "Edit" element. I would imagine that element will either take you to a different page where you do the editing or it acts like an interactive element that causes something to happen on the page. 

If it takes you to another page, it should be a link. If it performs an action on the page, it should be a button. That's what those HTML elements are created for. My guess is that yours should be a button. 

Of course, just because it's a button doesn't mean it has to _look_ like a button. You can style it to look more like a link or however you want. But my guess (with extremely limited knowledge of what you're trying to do) is that "Edit" should be a button (in which case `click_button` should work)

**Comment by petelacey** [comment]
To be clear, the example I used above is simply your README example with `click_link` changed to my proposed `click`. :smile: 

I'm totally on board with your semantic reasoning and have since changed the `div` element in question to an anchor element with no visual or functional changes.

Thanks for considering adding a generic `click` function.

**Comment by j4nk3e** [comment]
This function would be very helpful for me, too.

I have the issue which requires a generic `click` function: There is a table where each row is clickable to select it, so there is a `phx-click` on the `<tr>`. With `render_click` from `LiveViewTest` it works flawlessly, however I don't know how to implement this with `phoenix_test`.

Btw. I noticed there is an example using `click()` in the documentation, even though this function doesn't exist:
https://github.com/germsvel/phoenix_test/blob/3c601c4c63f64354fc950f961c5962a3d5c6fa9d/lib/phoenix_test.ex#L1347

**Comment by germsvel** [comment]
@j4nk3e thanks for commenting on this issue. 

I think you can accomplish what you want with what PhoenixTest has today. You could use `click_link/3` or `click_button/3` to specify the CSS selector you want to target, and it should let you click on that element. 

That being said, I always encourage people to think about what the element is trying to do in the page. I'm no accessibility expert, but I'm trying to learn more so that PhoenixTest does the "right" thing by default. I think if you want to click on a button that performs an action on the page, it should be a button-like element (links being more for navigation). 

To that end, I just merged a commit (1dec249eb5c02ee8235ef6f6ddb839a4c0bd17e6) that adds support for targeting elements with `role="button"`. So if you want to click a `<tr>` that performs an action on your page, the `main` branch should now allow you to target that `<tr role="button">` with `click_button/2`. 

Hope that helps!

P.S.: And if you're more curious about the button role, I was reading this (which I found helpful): https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role

**Comment by j4nk3e** [comment]
Thank you so much! I didn't know about the button role. That helped a lot, I can do everything I wanted with `phoenix_test` now.
